lib/models/per-bundle-addon-cache/index.js

```diff
@@ -1,10 +1,48 @@
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
 const isLazyEngine = require('../../utilities/is-lazy-engine');
 const { getAddonProxy } = require('./addon-proxy');
 const logger = require('heimdalljs-logger')('ember-cli:per-bundle-addon-cache');
 const { TARGET_INSTANCE } = require('./target-instance');
 
+function defaultAllowCachingPerBundle({ addonEntryPointModule }) {
+  return (
+    addonEntryPointModule.allowCachingPerBundle ||
+    (addonEntryPointModule.prototype && addonEntryPointModule.prototype.allowCachingPerBundle)
```

Comment:
Hmm, this seems wrong? `addonEntryPointModule` is the plain / non-"tweaked" export (usually `module.exports = { name: require('./package').name }`) now (it used to be the result of `getAddonConstructor()` but that was changed).

Reply:
I was mistaken here, the code is accounting for either case

@brendenpalmer is going to double check to ensure there are tests for each of these cases

Reply:
ah yeah, this case is just to cover if someone tries to opt-in as part of exporting a subclass of the base model addon directly; eg, `module.exports = Addon.extend({ allowCachingPerBundle: true });`. there wasn't a test-case to cover this, but i just updated this PR to test for this